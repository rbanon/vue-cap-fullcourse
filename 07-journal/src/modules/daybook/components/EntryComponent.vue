<template>
  <div
    class="entry-container mb-2 pointer p-2"
    @click="$router.push({ name: 'entry', params: { id: 10 } })"
  >
    <!-- Título-->
    <div class="entry-title d-flex">
      <span class="text-success fs-5 fw-bold">23</span>
      <span class="mx-1 fs-5">Agosto</span>
      <span class="mx-2 fw-light">2023, miércoles</span>
    </div>

    <div class="entry-description">
      Lorem, ipsum dolor sit amet consectetur adipisicing elit. Rem repudiandae
      magni distinctio sint possimus corrupti nemo excepturi rerum cumque
      delectus eligendi odio ab, laboriosam nesciunt iure pariatur veniam.
      Temporibus, recusandae.
    </div>
  </div>
</template>

<script>
export default {
  name: "EntryComponent",
};
</script>
<style lang="scss" scoped>
.entry-container {
  border-bottom: 1px solid #2c3e50;
  transition: 0.2s all ease-in-out;

  &:hover {
    background-color: lighten($color: grey, $amount: 45);
    transition: 0.2s all ease-in-out;
  }

  .entry-description {
    font-size: 12px;
  }
}
</style>
