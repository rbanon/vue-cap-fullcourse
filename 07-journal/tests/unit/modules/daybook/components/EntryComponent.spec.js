import { shallowMount } from "@vue/test-utils";
import EntryComponent from "@/modules/daybook/components/EntryComponent";
import { journalState } from "../../../mock-data/test-journal-state";

describe("Pruebas en el Entry Component", () => {
  const mockRouter = {
    push: jest.fn(),
  };

  const wrapper = shallowMount(EntryComponent, {
    props: {
      entry: journalState.entries[1],
    },
    global: {
      mocks: {
        $router: mockRouter,
      },
    },
  });

  it("Debe de coincidir con el del snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("Debe de redireccionar al hacer click en el entry-container", () => {
    const entryContainer = wrapper.find(".entry-container");
    entryContainer.trigger("click");

    expect(mockRouter.push).toHaveBeenCalledWith({
      name: "entry",
      params: { id: journalState.entries[1].id },
    });

    expect(mockRouter.push).toHaveBeenCalledTimes(1);
  });

  it("Pruebas en las propiedades computadas", () => {
    expect(wrapper.vm.day).toBe(1);
    expect(wrapper.vm.month).toBe("Septiembre");
    expect(wrapper.vm.yearDay).toBe("2023, Viernes");
  });
});
