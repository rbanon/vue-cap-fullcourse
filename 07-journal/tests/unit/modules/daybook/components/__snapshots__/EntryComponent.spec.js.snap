// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el Entry Component Debe de lcoincidir con el del snapshot 1`] = `
<div class="entry-container mb-2 pointer p-2">
  <!-- TÃ­tulo-->
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">1</span><span class="mx-1 fs-5">Septiembre</span><span class="mx-2 fw-light">2023, Viernes</span></div>
  <div class="entry-description">Prueba septiembre con imagen</div>
</div>
`;
